src/utils/handlers/general/searchTrack.ts

```diff
@@ -77,6 +77,9 @@ export async function searchTrack(
             case "youtube": {
                 switch (queryData.type) {
                     case "track": {
+                        if (/youtube\.com/g.exec(url.hostname)) {
+                            url = new URL(`https://youtu.be/${url.href.split('watch?v=')[1]}`);
+                        }
                         const track = await youtube.getVideo(
                             /youtu\.be/g.exec(url.hostname) ? url.pathname.replace("/", "") : url.toString()
                         );
```

Comment:
Instead of changing the `url` variable, you can just do this in the ternary:
```suggestion
                        const track = await youtube.getVideo(
                            /youtu\.be/g.exec(url.hostname) ? url.pathname.replace("/", "") : url.searchParams.get("v")
                        );
```